ioc_finder/ioc_finder.py

```diff
@@ -88,6 +88,22 @@ def prepare_text(text: str) -> str:
     return text
 
 
+def _clean_url(url: str) -> str:
+    """Clean the given URL, removing common, unwanted characters which are usually not part of the URL."""
+    # if there is a ")" in the URL and not a "(", remove everything including and after the ")"
+    if ")" in url and "(" not in url:
+        url = url.split(")")[0]
+
+    # remove `"` and `'` characters from the end of a URL
+    url = url.rstrip('"').rstrip("'")
+
+    # remove `'/>` and `"/>` from the end of a URL (this character string occurs at the end of an HMTL tag with )
+    url = string_remove_from_end(url, "'/>")
+    url = string_remove_from_end(url, '"/>')
+
+    return url
```

Comment:
Two changes to note:

1. Moving this into its own function for easier extensibility in the future
2. Removing `)` before rstriping quotation marks so cases like `http://example.com/foo')` are properly parsed as `http://example.com/foo`